### PR TITLE
Added DDRtest

### PR DESCRIPTION
**Pull request by Yuehan-Lin**
[ci skip]
Signed-off-by: Yuehan-Lin <Yuehan.Lin@ibm.com>

### CODE REVIEW

**Comment by llxia** [comment]
jenkins test sanity win jdk8, jdk11

**Comment by llxia** [comment]
jenkins test sanity win jdk8

**Comment by llxia** [comment]
jenkins test sanity win jdk11

**Comment by llxia** [comment]
jenkins test extended xlinux,plinux,aix,zlinux,xlinuxlargeheap jdk11

**Comment by llxia** [comment]
jenkins test sanity win jdk11

**Comment by llxia** [comment]
jenkins test extended xlinux,plinux,aix,zlinux,xlinuxlargeheap jdk11

File: test/functional/DDR_Test/src/j9vm/test/ddrext/junit/TestSharedClassesExt.java [hunk 1]
```diff
@@ -533,7 +540,7 @@ public void testShrcJITHForExt() {
 		}
 
 		for (int i = 0; i < lines.length; i++) {
-			if (lines[i].contains("java/lang/Object")) {
+			if (lines[i].contains("java/lang/Object") || lines[i].matches("\\s*r.*\\(\\).*j9rommethod .*")) {
```
Comment (keithc-ca) [inline test/functional/DDR_Test/src/j9vm/test/ddrext/junit/TestSharedClassesExt.java]: I think a comment explaining the second call to `matches` is appropriate here.

File: test/functional/DDR_Test/playlist.xml [hunk 2]
```diff
@@ -120,18 +119,136 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
 		<testCaseName>testDDRExt_Callsites_win</testCaseName>
```
[the remark this answers is not in the document]
Comment (llxia) [reply test/functional/DDR_Test/playlist.xml]: We mark test on Windows as part of sanity group and the non-Windows for the extended group.

[the remark this answers is not in the document]
Comment (keithc-ca) [reply test/functional/DDR_Test/playlist.xml]: Ok, but why is Windows treated specially with respect to this test?

[the remark this answers is not in the document]
Comment (smlambert) [reply test/functional/DDR_Test/playlist.xml]: It is most likely treated like that because we carried over the levels directly as they were from the old internal system that would have tagged that platform as 'sanity' / gold / blocking.  There is no way for us to know 'why' that would have been done.  

If I were to guess, it was likely done to choose 1 platform out of the bunch to watch to for failures (and that could block a build), with the other platforms moved out to extended, to keep the sanity execution time shorter.  Since we now look at both sanity & extended failures more closely in the openj9 project, I think it is fine to collapse all platforms into 1 test target tagged as extended.

**Comment by Yuehan-Lin** [comment]
@keithc-ca please review, thank you!

File: test/functional/DDR_Test/ddrSettings.mk [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+#
+#  SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+##############################################################################
+
+ifeq ("$(JDK_VERSION)","8")
+export ADDEXPORTS_JDKASM_UNNAMED=-showversion
+else
+export ADDEXPORTS_JDKASM_UNNAMED=--add-exports=java.base/jdk.internal.org.objectweb.asm=ALL-UNNAMED
+endif
+
```
Comment (keithc-ca) [inline test/functional/DDR_Test/ddrSettings.mk]: I had asked for a final newline, not an extra blank line. ;-)

File: test/functional/DDR_Test/playlist.xml [hunk 4]
```diff
@@ -14,18 +14,17 @@ forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
 General Public License, version 2 with the GNU Classpath
 Exception [1] and GNU General Public License, version 2 with the
 OpenJDK Assembly Exception [2].
-
```
Comment (keithc-ca) [inline test/functional/DDR_Test/playlist.xml]: Please restore the blank lines that separate paragraphs of the copyright notice.

File: test/functional/DDR_Test/playlist.xml [hunk 5]
```diff
@@ -57,13 +56,14 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
 		<impls>
 			<impl>openj9</impl>
 		</impls>
+		<disabled>https://github.com/eclipse/openj9/issues/5089</disabled>
```
Comment (keithc-ca) [inline test/functional/DDR_Test/playlist.xml]: The test should be fixed rather than excluded.

File: test/functional/DDR_Test/playlist.xml [hunk 6]
```diff
@@ -129,9 +130,127 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
...
+
+	<test>
+		<testCaseName>testDDRExt_JITExt_win</testCaseName>
+		<command>ant -v -DJAVA_COMMAND=$(JAVA_COMMAND) -DTEST_ROOT=${TEST_ROOT} -DJDK_HOME=${JDK_HOME} -DJDK_VERSION=${JDK_VERSION} \
+		-DTEST_RESROOT=$(TEST_RESROOT) -DRESOURCES_DIR=${RESOURCES_DIR} -DREPORTDIR=${REPORTDIR} -DOS=${OS} -DBITS=$(BITS) \
+		-Dtest.list=$(Q)TestJITExt$(Q) -DADDITIONALEXPORTS=$(ADDEXPORTS_JDKASM_UNNAMED) -DEXTRADUMPOPT=$(Q)-Xjit:count=0$(Q) -f $(Q)$(TEST_RESROOT)$(D)tck_ddrext.xml$(Q); \
+		$(TEST_STATUS)</command>
+		<platformRequirements>os.win</platformRequirements>
+		<levels>
+			<level>sanity</level>
```
Comment (keithc-ca) [inline test/functional/DDR_Test/playlist.xml]: Didn't we agree that the duplication for Windows would be removed?

File: test/functional/DDR_Test/playlist.xml [hunk 7]
```diff
@@ -129,9 +130,127 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
...
+		</groups>
+		<impls>
 			<impl>ibm</impl>
+			<impl>openj9</impl>
 		</impls>
+		<disabled>https://github.com/eclipse/openj9/issues/5089</disabled>
 	</test>
-</playlist>
+
+</playlist>
```
Comment (keithc-ca) [inline test/functional/DDR_Test/playlist.xml]: Please include a final newline in all files.

File: test/functional/DDR_Test/src/j9vm/test/ddrext/Constants.java [hunk 8]
```diff
@@ -258,18 +258,18 @@
 
 	public static final String SHRC_FINDJITH = "findjith";
 	public static final String SHRC_FINDJITH_METHODNAME = "run";
-	public static final String SHRC_FINDJITH_SUCCESS_KEY = "run\\(\\)Ljava/lang/.*,JITHINT data !j9x,.*/Object";
-	public static final String SHRC_FINDJITH_FAILURE_KEY = "No entry found in the cache";
+	public static final String SHRC_FINDJITH_SUCCESS_KEY = "((run|resize)\\(\\)(Ljava/lang/.*)?|No entry found in the cache)";
```
Comment (keithc-ca) [inline test/functional/DDR_Test/src/j9vm/test/ddrext/Constants.java]: Why is 'No entry found in the cache' no longer considered a failure?

File: test/functional/DDR_Test/src/j9vm/test/ddrext/junit/TestSharedClassesExt.java [hunk 9]
```diff
@@ -519,9 +519,6 @@ public void testShrcJITHForExt() {
 		String findjitpOutput = exec(Constants.SHRC_CMD, new String[] {
 				Constants.SHRC_FINDJITH, Constants.SHRC_FINDJITH_METHODNAME });
 		if (findjitpOutput.contains("JITHINT data") == false) {
-			fail("No jit hint information found for the method :"
-					+ Constants.SHRC_FINDJITH_METHODNAME
-					+ ". Can not proceed with test");
```
Comment (keithc-ca) [inline test/functional/DDR_Test/src/j9vm/test/ddrext/junit/TestSharedClassesExt.java]: Why is this no longer a failure?

File: test/functional/DDR_Test/src/j9vm/test/ddrext/junit/TestSharedClassesExt.java (hunk 1 above)
Comment (keithc-ca) [reply test/functional/DDR_Test/src/j9vm/test/ddrext/junit/TestSharedClassesExt.java]: The comment added doesn't help much: can you elaborate?

File: test/functional/DDR_Test/tck_ddrext.xml [hunk 10]
```diff
@@ -113,13 +149,98 @@
...
+		<echo>Setting up TCK</echo>
+		<mkdir dir="${tck.output.dir}"/>
+		<mkdir dir="${tck.output.dir}_${tck.type}"/>
+		<echo>Recording Java version</echo>
+		<java fork="true" jvm="${sdk.executable}${sdk.executable.ext}" classname="com.ibm.dtfj.tck.harness.TCKLauncher" 
+			failonerror="true" outputproperty="jvm.version" newenvironment="true">
+			<jvmarg value="-version" />
+		</java>
+		<condition property="java.is64bit">
+			<matches pattern=" jvm\S+64| p\S+64|\-64 " string="${jvm.version}" multiline="true"/>
```
Comment (keithc-ca) [inline test/functional/DDR_Test/tck_ddrext.xml]: `-` does not need to be escaped in this pattern; `\-64` can be just `-64`.

File: test/functional/DDR_Test/tck_ddrext.xml [hunk 11]
```diff
@@ -113,13 +149,98 @@
...
+
+		<if>
+			<isset property="isZOS"/>
+			<then>
+				<var name="temp.tck.config.dump.opt" value="-Xdump:system:defaults:request=exclusive+compact+prepwalk,label=${user.name}.${tck.core.file.mvs}"/>
+			</then>
+			<elseif>
+				<isset property="isWindows"/>
+				<then>
+					<tempfile property="path.file"/>
```
Comment (keithc-ca) [inline test/functional/DDR_Test/tck_ddrext.xml]: A file is not needed for this; `<pathconvert/>` is probably what you want here.
See https://ant.apache.org/manual/Tasks/pathconvert.html

**Comment by llxia** [comment]
DDR is supported on osx (PR https://github.com/eclipse/openj9/pull/5127), please enable DDR tests on osx.

**Comment by Yuehan-Lin** [comment]
@keithc-ca ready for review

File: test/functional/DDR_Test/playlist.xml [hunk 12]
```diff
@@ -18,16 +18,16 @@ OpenJDK Assembly Exception [2].
 [1] https://www.gnu.org/software/classpath/license.html
 [2] http://openjdk.java.net/legal/assembly-exception.html
 
-SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+  SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
```
Comment (keithc-ca) [inline test/functional/DDR_Test/playlist.xml]: Please undo this change.

File: test/functional/DDR_Test/src/j9vm/test/corehelper/CoreGen.java [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+
+ public class CoreGen {
+
+ 	public static void main(String[] args) {
+		SimpleThread st = new SimpleThread();
+		st.configureObj();
+		st.configureValidJavaThreads();
+	}
+
+ }
```
Comment (keithc-ca) [inline test/functional/DDR_Test/src/j9vm/test/corehelper/CoreGen.java]: Please fix the indentation and remove the last line (which has a space but no new-line).

File: test/functional/DDR_Test/src/j9vm/test/ddrext/Constants.java [hunk 14]
```diff
@@ -642,4 +641,4 @@
 	
 	public static final String ROMCLASS_FOR_NAME_CMD = "romclassforname";
 	
-}
+}
```
Comment (keithc-ca) [inline test/functional/DDR_Test/src/j9vm/test/ddrext/Constants.java]: Please restore the new-line at the end of this (and every) file.

File: test/functional/DDR_Test/src/j9vm/test/ddrext/junit/TestSharedClassesExt.java [hunk 15]
```diff
@@ -533,7 +533,8 @@ public void testShrcJITHForExt() {
 		}
 
 		for (int i = 0; i < lines.length; i++) {
-			if (lines[i].contains("java/lang/Object")) {
+			if (lines[i].contains("java/lang/Object") || lines[i].matches("\\s<init>\\(\\).*j9rommethod .*")) {
+				// lines[i].matches: find j9rommethod address from <init> method in openj9
```
Comment (keithc-ca) [inline test/functional/DDR_Test/src/j9vm/test/ddrext/junit/TestSharedClassesExt.java]: Please improve this comment. It should help the reader understand *why* the condition is what it is, not merely repeat it.

File: test/functional/DDR_Test/tck_ddrext.xml [hunk 16]
```diff
@@ -113,13 +144,91 @@
 	</target>
 
 	<target name="TCK.move.dump.ddrext" description="Move core file from MVS to HFS">
-			<echo>Moving core file from MVS [${dump.name}${zos.core.suffix}] to HFS [${system.dump}]</echo>
-			<exec executable="/bin/mv" failonerror="true">
-				<arg line="&quot;//${dump.name}${zos.core.suffix}&quot; ${system.dump}"/>
-			</exec>
+		<echo>Moving core file from MVS [${dump.name}${zos.core.suffix}] to HFS [${system.dump}]</echo>
+		<exec executable="/bin/mv" failonerror="true">
+			<arg line="&quot;//${dump.name}${zos.core.suffix}&quot; ${system.dump}"/>
```
Comment (keithc-ca) [inline test/functional/DDR_Test/tck_ddrext.xml]: This will be more robust written as two `<arg value="..."/>` elements.

File: test/functional/DDR_Test/tck_ddrext.xml (hunk 11 above)
Comment (Yuehan-Lin) [reply test/functional/DDR_Test/tck_ddrext.xml]: I changed into `<pathconvert>` now, could you please take a look?

Comment (Yuehan-Lin) [reply test/functional/DDR_Test/tck_ddrext.xml]: I changed into `<pathconvert>` now, could you please take a look? @keithc-ca

Comment (keithc-ca) [reply test/functional/DDR_Test/tck_ddrext.xml]: It's actually even simpler if we use `<property ... location="..."/>` which does the right thing on Windows and other platforms:
```
<if>
	<isset property="isZOS" />
	<then>
		<property name="system.dump.location" value="${user.name}.${tck.core.file.mvs}" />
	</then>
	<else>
		<property name="system.dump.location" location="${system.dump}" />
	</else>
</if>
<property name="tck.config.dump.opt.1" value="-Xdump:system:defaults:label=${system.dump.location},request=exclusive+compact+prepwalk" />
```

File: test/functional/DDR_Test/tck_ddrext.xml [hunk 17]
```diff
@@ -118,11 +148,92 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
...
+		<mkdir dir="${tck.output.dir}"/>
+		<mkdir dir="${tck.output.dir}_${tck.type}"/>
+		<echo>Recording Java version</echo>
+		<java fork="true" jvm="${sdk.executable}${sdk.executable.ext}" classname="com.ibm.dtfj.tck.harness.TCKLauncher" 
+			failonerror="true" outputproperty="jvm.version" newenvironment="true">
+			<jvmarg value="-version" />
+		</java>
+		<condition property="java.is64bit">
+			<matches pattern=" jvm\S+64| p\S+64|-64 " string="${jvm.version}" multiline="true"/>
+		</condition>
```
Comment (keithc-ca) [inline test/functional/DDR_Test/tck_ddrext.xml]: I think this would be more reliable using `java -XshowSettings:properties -version` looking for `sun.arch.data.model = 64`.

File: test/functional/DDR_Test/tck_ddrext.xml (hunk 11 above)
Comment (keithc-ca) [reply test/functional/DDR_Test/tck_ddrext.xml]: Actually, because `ddrext.setup.dump.dynamic` sets `system.dump` properly on z/OS, this doesn't need to make a special case and can be further simplified:
```
<property name="system.dump.location" location="${system.dump}" />
<property name="tck.config.dump.opt.1" value="-Xdump:system:defaults:label=${system.dump.location},request=exclusive+compact+prepwalk" />
```

File: test/functional/DDR_Test/tck_ddrext.xml (hunk 17 above)
Comment (Yuehan-Lin) [reply test/functional/DDR_Test/tck_ddrext.xml]: I changed`<property ... location="..."/>` and `sun.arch.data.model = 64` part. Please review again @keithc-ca

File: test/functional/DDR_Test/playlist.xml (hunk 5 above)
Comment (keithc-ca) [reply test/functional/DDR_Test/playlist.xml]: Why is this still disabled?

File: test/functional/DDR_Test/src/j9vm/test/ddrext/Constants.java [hunk 18]
```diff
@@ -257,18 +257,18 @@
 	public static final String SHRC_JITHSTATS_FAILURE_KEY = "no shared cache,JITPROFILE data length 0 metadata 0";
 
 	public static final String SHRC_FINDJITH = "findjith";
-	public static final String SHRC_FINDJITH_METHODNAME = "run";
-	public static final String SHRC_FINDJITH_SUCCESS_KEY = "run\\(\\)Ljava/lang/.*,JITHINT data !j9x,.*/Object";
+	public static final String SHRC_FINDJITH_METHODNAME = "<init>";
+	public static final String SHRC_FINDJITH_SUCCESS_KEY = "<init>\\(.*\\)";
```
Comment (keithc-ca) [inline test/functional/DDR_Test/src/j9vm/test/ddrext/Constants.java]: Why was `JITHINT data !j9x,.*/Object` removed here and in other success keys below?

File: test/functional/DDR_Test/tck_ddrext.xml [hunk 19]
```diff
@@ -76,7 +81,7 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
 			<echo>Using JVM : ${JAVA_COMMAND}</echo>
 			<echo>classname = "j9vm.test.corehelper.StackMapCoreGenerator"</echo>
 			<echo>Java VM Args:</echo>
-			<echo>  jvmarg = -Xshareclasses:name=ddrextjunitSCC,destroy</echo>
+			<echo>	jvmarg = -Xshareclasses:name=ddrextjunitSCC,destroy</echo>
```
Comment (keithc-ca) [inline test/functional/DDR_Test/tck_ddrext.xml]: Please don't embed tabs; they should only be used for indentation.

File: test/functional/DDR_Test/src/j9vm/test/ddrext/Constants.java (hunk 18 above)
Comment (Yuehan-Lin) [reply test/functional/DDR_Test/src/j9vm/test/ddrext/Constants.java]: I added them back now, please take a look @keithc-ca

Comment (Yuehan-Lin) [reply test/functional/DDR_Test/src/j9vm/test/ddrext/Constants.java]: Ready for review, please take a look @keithc-ca

File: test/functional/DDR_Test/tck_ddrext.xml [hunk 20]
```diff
@@ -43,8 +48,8 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
 	<if>
 		<equals arg1="${OS}" arg2="os.zos"/>
 		<then>
-			<property name="DUMP_OPTION" value="-Xdump:system:events=vmstop,label=%uid.${dump.name},request=exclusive+compact+prepwalk -Xscmx16m -Xshareclasses:name=ddrextjunitSCC,addtestjithints -Xjit -Xaot:forceaot,count=5,disableAsyncCompilation -Xmx512M -version" />
-				<if>
+			<property name="DUMP_OPTION" value="-Xdump:system:defaults:file=${system.dump},request=exclusive+compact+prepwalk -Xscmx16m -Xshareclasses:name=ddrextjunitSCC,addtestjithints -Xjit -Xaot:forceaot,count=5,disableAsyncCompilation -Xmx512M" />
```
Comment (keithc-ca) [inline test/functional/DDR_Test/tck_ddrext.xml]: This needs to use `label=` (or `dsn=`, not `file=`); z/OS does not support the `file=` sub-option.

Comment (keithc-ca) [reply test/functional/DDR_Test/tck_ddrext.xml]: @Yuehan-Lin I believe this is the final point that I asked to be addressed: will you be pushing a change soon?

Comment (Yuehan-Lin) [reply test/functional/DDR_Test/tck_ddrext.xml]: Sure, I just pushed. I was just testing on vmfarm. And for my latest change, I changed `label=${system.dump}`back into `label=%uid.${dump.name}` in `test/functional/DDR_Test/tck_ddrext.xml`. 
This is because when I tested on vmfarm, teston zos gave the error
 ```
[exec] IEATDUMP failure for DSN='/HOME/J9BUILD/TMP/BLD_413603/TESTDDREXT_GENERAL_IBM_0/DDREXT.J9CORE.DMP.X&DS' RC=0x00000008 RSN=0x00000022
 [exec] JVMPORT023W IEATDUMP failed because user-specified dump template was too long. Retrying dump with default template.
```

Comment (keithc-ca) [reply test/functional/DDR_Test/tck_ddrext.xml]: The special treatment of the definition of `DUMP_OPTION` afforded on z/OS will also need to be applied in `ddrext.setup.dump.dynamic`.

Comment (Yuehan-Lin) [reply test/functional/DDR_Test/tck_ddrext.xml]: I am not sure if this is correctly: do you mean that for `tck.config.dump.opt.1`, I should change it back to using `temp.tck.config.dump.opt` to discuss if os is zos or windows platform, then eventually set
```
<property name="tck.config.dump.opt.1" value="${temp.tck.config.dump.opt}"/>
```

Comment (keithc-ca) [reply test/functional/DDR_Test/tck_ddrext.xml]: What I mean is that we need to effectively replace
```
<property name="system.dump.location" location="${system.dump}" />
```
with
```
<if>
  <isset property="isZOS" />
  <then>
    <property name="system.dump.location" value="${dump.name}${zos.core.suffix}" />
  </then>
  <else>
    <property name="system.dump.location" location="${system.dump}" />
  </else>
</if>
```
to avoid the path that is too long on z/OS.

Comment (Yuehan-Lin) [reply test/functional/DDR_Test/tck_ddrext.xml]: I just changed and pushed, please take a look

Comment (keithc-ca) [reply test/functional/DDR_Test/tck_ddrext.xml]: That looks good. Have you confirmed that it addresses the problem you saw on z/OS?

Comment (Yuehan-Lin) [reply test/functional/DDR_Test/tck_ddrext.xml]: Not yet, it's still testing now. I will let you know the result once it's finished. Thank you!

Comment (Yuehan-Lin) [reply test/functional/DDR_Test/tck_ddrext.xml]: I changed playlist.xml to exclude `testDDRExt_General_ibm` on zos temporarily, since !vmcheck failed on zos platform, (runtimes/backlog/issues/124). And the rest ddr tests passed.

**Comment by keithc-ca** [comment]
Jenkins test extended aix,osx,plinux,xlinux,zlinux,win jdk8,jdk11

**Comment by Yuehan-Lin** [comment]
Last pr test failure is fixed, please review and test @keithc-ca

**Comment by keithc-ca** [comment]
Jenkins test extended plinux jdk8,jdk11

File: test/functional/DDR_Test/tck_ddrext.xml [hunk 21]
```diff
@@ -102,11 +107,40 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
...
 		<echo>Generate dump file</echo>
+		<echo>JAVA_COMMAND: ${JAVA_COMMAND}</echo>
+		<echo>DUMP_OPTION: ${DUMP_OPTION}</echo>
+		<echo>EXTRADUMPOPT: ${EXTRADUMPOPT}</echo>
+
 		<exec executable="${JAVA_COMMAND}" failonerror="true">
 			<arg line="${DUMP_OPTION}" />
+			<arg line="-Xitsn1000" />
+			<arg line="${EXTRADUMPOPT}" />
+			<arg value="-Xjit:{j9vm/test/corehelper/CoreGen.main*}(count=0)"/>
```
Comment (llxia) [inline test/functional/DDR_Test/tck_ddrext.xml]: Is it possible to use `${EXTRADUMPOPT}` for passing in these two options (-Xitsn1000 and -Xjit...)?

Comment (keithc-ca) [reply test/functional/DDR_Test/tck_ddrext.xml]: I don't think that's worth it: we'd have to be careful about quoting or otherwise ensuring that `<arg line=...>` properly separates arguments - a problem that doesn't arise with `<arg value=...>`.

**Comment by keithc-ca** [comment]
Jenkins test extended aix,osx,xlinux,zlinux,win jdk8,jdk11

File: test/functional/DDR_Test/tck_ddrext.xml (hunk 21 above)
Comment (llxia) [reply test/functional/DDR_Test/tck_ddrext.xml]: Yes, quoting and special characters may be a problem.

**Comment by keithc-ca** [comment]
The AIX failure seems unrelated to this and more likely related to #3081.

**Comment by llxia** [comment]
@Yuehan-Lin  Since runtimes/backlog/issues/124 is fixed, please re-enable zos in a separate PR.
